clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -92,6 +92,20 @@ def CIR_BoolAttr : CIR_Attr<"Bool", "bool", [TypedAttrInterface]> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// PlaceholderAttr
+//===----------------------------------------------------------------------===//
+
+def PlaceholderAttr : CIR_Attr<"Placeholder", "placeholder", [TypedAttrInterface]> {
+  let summary = "Attribute to represent a placeholder";
```

Comment:
Let's call this `UninitializedUnionField`. If something else shows up that needs this we can iterate on a more generic name.